msg.go

```diff
@@ -732,12 +732,9 @@ func (dns *Msg) PackBuffer(buf []byte) (msg []byte, err error) {
 
 	// We need the uncompressed length here, because we first pack it and then compress it.
 	msg = buf
-	compress := dns.Compress
```

Comment:
it's possible. let's say you implement a caching DNS server that caches a DNS response, and you serve up two requests for the same domain (which has been cached) at roughly the same time.